File: dotNET/PFProSDK/src/PayPal/Payments/DataObjects/ECUpdateBARequest.cs
#region "Imports"

using System;
using PayPal.Payments.Common.Utility ;  
using PayPal.Payments.Common; 
#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// Used for ExpressCheckout with Billing Agreement (Reference Transaction) without Purchase DO operation.
	/// </summary>
	/// <remarks>
	/// <seealso cref="ECGetBARequest"/>
	/// <seealso cref="ECSetBARequest"/>
  /// <seealso cref="ECDoBARequest"/>
	/// </remarks>
	public class ECUpdateBARequest : ExpressCheckoutRequest
	{
		#region "Member Variables"
		private String mBAId;
		private String mBA_Status;
		private String mBA_Desc;
		#endregion

		#region "Properties"
		
		/// <summary>
		/// Gets or Sets the ba_status parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BA_STATUS</code>
		/// </remarks>
		public String BA_Status
		{
			get{return mBA_Status; }
			set{mBA_Status = value;}
		}
		/// <summary>
		/// Gets or Sets the baid parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BAID</code>
		/// </remarks>
		public String BAId
		{
			get{return mBAId; }
			set{mBAId = value;}
		}
		/// <summary>
		/// Gets or Sets the ba_desc parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BA_DESC</code>
		/// </remarks>
		public String BA_Desc
		{
			get{return mBA_Desc; }
			set{mBA_Desc = value;}
		}

		#endregion

		#region "Constructor"

		/// <summary>
		/// Constructor for ECDoBARequest
		/// </summary>
		/// <param name="BAId">String</param>
		/// <remarks>
		/// ECDoBARequest is used to set the data required for a Express Checkout UPDATE operation
		/// with Billing Agreement (Reference Transaction) without Purchase.
		/// </remarks>
		/// <example>
		/// <code lang="C#" escaped="false">
		///		.............
		///		
		///		//Create the ECUpdateBARequest object
		///		ECUpdateBARequest UpdateEC = new ECUpdateBARequest("[baid]");
		///		
		///		.............
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		///		.............
		///		
		///		//Create the ECUpdateBARequest object
		///		Dim UpdateEC As ECUpdateBARequest = new ECUpdateBARequest("[baid]")
		///		
		///		.............
		/// </code>
		/// </example>
		public ECUpdateBARequest(String BAId) : base(PayflowConstants.ACTIONTYPE_UPDATEBA)
		{
			mBAId = BAId;
		}
		/// <summary>
		/// Constructor for ECDoBARequest
		/// </summary>
		/// <param name="BAId">String</param>
		/// <param name="BA_Status">String</param>
		/// <remarks>
		/// ECDoBARequest is used to set the data required for a Express Checkout UPDATE operation
		/// with Billing Agreement (Reference Transaction) without Purchase.
		/// </remarks>
		/// <example>
		/// <code lang="C#" escaped="false">
		///		.............
		///		
		///		//Create the ECUpdateBARequest object
		///		ECUpdateBARequest UpdateEC = new ECUpdateBARequest("[baid]", "[ba_status]");
		///		
		///		.............
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		///		.............
		///		
		///		//Create the ECUpdateBARequest object
		///		Dim UpdateEC As ECUpdateBARequest = new ECUpdateBARequest("[baid]", "[ba_status]")
		///		
		///		.............
		/// </code>
		/// </example>
		public ECUpdateBARequest(String BAId, String BA_Status) : base(PayflowConstants.ACTIONTYPE_UPDATEBA)
		{
			mBAId = BAId;
			mBA_Status = BA_Status;			
		}
		/// <summary>
		/// Constructor for ECDoBARequest
		/// </summary>
		/// <param name="BAId">String</param>
		/// <param name="BA_Status">String</param>
		/// <param name="BA_Desc">String</param>
		/// <remarks>
		/// ECDoBARequest is used to set the data required for a Express Checkout UPDATE operation
		/// with Billing Agreement (Reference Transaction) without Purchase.
		/// </remarks>
		/// <example>
		/// <code lang="C#" escaped="false">
		///		.............
		///		
		///		//Create the ECUpdateBARequest object
		///		ECUpdateBARequest UpdateEC = new ECUpdateBARequest("[baid]", "[ba_status]", ["ba_desc"]);
		///		
		///		.............
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		///		.............
		///		
		///		//Create the ECUpdateBARequest object
		///		Dim UpdateEC As ECUpdateBARequest = new ECUpdateBARequest("[baid]", "[ba_status]", ["ba_desc"])
		///		
		///		.............
		/// </code>
		/// </example>
		public ECUpdateBARequest(String BAId, String BA_Status, String BA_Desc) : base(PayflowConstants.ACTIONTYPE_UPDATEBA)
		{
			mBAId = BAId;
			mBA_Status = BA_Status;
			mBA_Desc = BA_Desc;
		}
		#endregion

		#region "Core functions"

		/// <summary>
		/// Generates the transaction request.
		/// </summary>
		internal override void GenerateRequest()
		{
			// This function is not called. All the
			//address information is validated and generated
			//in its respective derived classes.
			base.GenerateRequest ();
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_BA_STATUS, mBA_Status));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_BA_DESC, mBA_Desc));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_BAID, mBAId));
		}
		#endregion
	}
}

File: java/src/sdk/dataobjects/paypal/payflow/ECUpdateBARequest.java
package paypal.payflow;

/**
 * Used for ExpressCheckout UPDATE operation.
 * {@link ECGetRequest}
 * {@link ECDoRequest}
 */
public class ECUpdateBARequest extends ExpressCheckoutRequest {

    private String baid;
    private String ba_Status;
    private String ba_Desc;

    /**
     * Constructor for ECSetRequest
     *
     * @param BAId - String
     *             <p/>
     *             ECUpdateRequest is used to set the data required for a Express Checkout UPDATE operation.
     *             </p>
     * @paypal.sample .............
     * <p/>
     * //Create the ECUpdateRequest object
     * ECUpdateRequest updateEC = new ECUpdateRequest("baid");
     * <p/>
     * .............
     * </code>
     */
    public ECUpdateBARequest(String BAId) {
        super(PayflowConstants.PARAM_ACTION_UPDATE);
        baid = BAId;
    }

    /**
     * Constructor for ECUpdateRequest
     *
     * @param BAId      - String
     * @param BA_Status - String
     *                  <p/>
     *                  <p/>
     *                  ECSetRequest is used to set the data required for a Express Checkout Update operation for
     *                  Reference Transactions without Purchase.
     *                  </p>
     * @paypal.sample .............
     * <p/>
     * //Create the ECUpdateRequest object
     * ECUpdateRequest updateEC = new ECSetRequest("baid","ba_status");
     * <p/>
     * .............
     * </code>
     */
    public ECUpdateBARequest(String BAId, String BA_Status) {
        super(PayflowConstants.PARAM_ACTION_UPDATE);
        baid = BAId;
        ba_Status = BA_Status;
    }

    /**
     * Constructor for ECUpdateRequest
     *
     * @param BAId      - String
     * @param BA_Status - String
     * @param BA_Desc   - String
     *                  <p/>
     *                  <p/>
     *                  ECSetRequest is used to set the data required for a Express Checkout Update operation for
     *                  Reference Transactions without Purchase.
     *                  </p>
     * @paypal.sample .............
     * <p/>
     * //Create the ECUpdateRequest object
     * ECUpdateRequest updateEC = new ECSetRequest("baid","ba_status", "ba_desc");
     * <p/>
     * .............
     * </code>
     */
    public ECUpdateBARequest(String BAId, String BA_Status, String BA_Desc) {
        super(PayflowConstants.PARAM_ACTION_UPDATE);
        baid = BAId;
        ba_Status = BA_Status;
        ba_Desc = BA_Desc;
    }

    /**
     * Gets the BAId parameter.
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: BAID
     */
    public String getBAId() {
        return baid;
    }

    /**
     * Sets the BAId parameter.
     *
     * @param BAId - String
     * @paypal.sample <p>Maps to Payflow Parameter: BAID
     */

    public void setBAId(String BAId) {
        this.baid = BAId;
    }

    /**
     * Gets the BA_Status parameter.
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: BA_STATUS
     */
    public String getba_Status() {
        return ba_Status;
    }

    /**
     * Sets the BA_Status parameter.
     *
     * @param BA_Status - String
     * @paypal.sample <p>Maps to Payflow Parameter: BA_STATUS
     */

    public void setba_Status(String BA_Status) {
        this.ba_Status = BA_Status;
    }

    /**
     * Gets the BA_Desc parameter.
     *
     * @return - String
     * @paypal.sample <p>Maps to Payflow Parameter: BA_DESC
     */
    public String getBA_Desc() {
        return ba_Desc;
    }

    /**
     * Sets the BA_Desc parameter.
     *
     * @param BA_Desc - String
     * @paypal.sample <p>Maps to Payflow Parameter: BA_DESC
     */

    public void setBA_Desc(String BA_Desc) {
        this.ba_Desc = BA_Desc;
    }

    protected void generateRequest() {
        // This function is not called. All the
        //address information is validated and generated
        //in its respective derived classes.
        super.generateRequest();
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_BA_STATUS, ba_Status));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_BA_DESC, ba_Desc));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_BAID, baid));

    }
}

